### PR TITLE
chore: consolidate environment variables

### PR DESCRIPTION
**Pull request by thsig**
**What this PR does / why we need it**:

We now read and export most environment variables used by the framework in `constants.ts`, using the `env-var` library.

This should simplify tracking the environment variables used by the framework (compared to ad-hoc reading of `process.env.*`).

### CODE REVIEW

File: garden-service/src/constants.ts [hunk 1]
```diff
@@ -42,3 +43,66 @@ export const SEGMENT_PROD_API_KEY = "b6ovUD9A0YjQqT3ZWetWUbuZ9OmGxKMa"
...
+    .asString(),
+  GARDEN_DISABLE_ANALYTICS: env
+    .get("GARDEN_DISABLE_ANALYTICS")
+    .required(false)
+    .asBool(),
+  GARDEN_DISABLE_VERSION_CHECK: env
+    .get("GARDEN_DISABLE_VERSION_CHECK")
+    .required(false)
+    .asString(),
+  GARDEN_ENABLE_PROFILING: env
```
Comment (eysi09) [inline garden-service/src/constants.ts:78]: Should we use `asBool` here? It coerces `"1"` and `"0"` to  `true` and `false` respectively.

File: garden-service/src/cli/cli.ts [hunk 2]
```diff
@@ -526,7 +532,7 @@ export async function run(): Promise<void> {
     console.log(err)
     code = 1
   } finally {
-    if (process.env.GARDEN_ENABLE_PROFILING === "1") {
+    if (gardenEnv.GARDEN_ENABLE_PROFILING === "1") {
```
Comment (eysi09) [inline garden-service/src/cli/cli.ts:535]: We can skip the comparison if we treat this as bool, see my comment below.

File: garden-service/src/constants.ts [hunk 3]
```diff
@@ -42,3 +43,66 @@ export const SEGMENT_PROD_API_KEY = "b6ovUD9A0YjQqT3ZWetWUbuZ9OmGxKMa"
...
+    .asString(),
+  GARDEN_LOG_LEVEL: env
+    .get("GARDEN_LOG_LEVEL")
+    .required(false)
+    .asString(),
+  GARDEN_LOGGER_TYPE: env
+    .get("GARDEN_LOGGER_TYPE")
+    .required(false)
+    .asString(),
+  GARDEN_SERVER_PORT: env
```
Comment (eysi09) [inline garden-service/src/constants.ts:90]: We can use `asPortNumber` here.

File: garden-service/src/server/server.ts [hunk 4]
```diff
@@ -45,7 +45,7 @@ export async function startServer(log: LogEntry, port?: number) {
   // Start HTTP API and dashboard server.
   // allow overriding automatic port picking
   if (!port) {
-    port = Number(process.env.GARDEN_SERVER_PORT) || undefined
+    port = Number(gardenEnv.GARDEN_SERVER_PORT) || undefined
```
Comment (eysi09) [inline garden-service/src/server/server.ts:48]: We can skip `Number()` if we coerce this to port number, see my comment above.

**Comment by thsig** [comment]
I think I've addressed all the comments now. If this is good to go, I'll squash before merging.

**Pull request merged**